Bottom: 48f97bb36e4ec8c69139aeacb0a6d53e6c29b7c1
Top:    5b69745c82ff50a573aadee813dd9eecd9a4f378
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-06 14:59:07 +0000

Try to prepend, not append to g_lists. This avoids wasting time traversing lists


---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index 7def7f2..c24ab8a 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -119,19 +119,19 @@ object3d_set_appearance (object3d *object, appearance *appear)
 void
 object3d_add_edge (object3d *object, edge_ref edge)
 {
-  object->edges = g_list_append (object->edges, (void *)edge);
+  object->edges = g_list_prepend (object->edges, (void *)edge);
 }
 
 void
 object3d_add_vertex (object3d *object, vertex3d *vertex)
 {
-  object->vertices = g_list_append (object->vertices, vertex);
+  object->vertices = g_list_prepend (object->vertices, vertex);
 }
 
 void
 object3d_add_face (object3d *object, face3d *face)
 {
-  object->faces = g_list_append (object->faces, face);
+  object->faces = g_list_prepend (object->faces, face);
 }
 
 
@@ -696,7 +696,7 @@ object3d_from_contours (POLYAREA *contours,
         }
 #endif
 
-      objects = g_list_append (objects, object);
+      objects = g_list_prepend (objects, object);
 
       link = malloc (sizeof (polygon_3d_link));
       pa->user_data = link;
